Editor/API/Solver/PluginResolver.cs

```diff
@@ -96,14 +96,14 @@ public static IEnumerable<Type> FindPluginTypes() => AppDomain.CurrentDomain.Get
         private static IPluginInternal? InstantiatePlugin(Type pluginType) =>
             pluginType.GetConstructor(Type.EmptyTypes)?.Invoke(Array.Empty<object>()) as IPluginInternal;
 
-        public static IEnumerable<IPluginInternal?> FindAllPlugins() => FindPluginTypes().Select(InstantiatePlugin);
+        public static IEnumerable<IPluginInternal> FindAllPlugins() => FindPluginTypes().Select(InstantiatePlugin).Where(p => p != null)!;
 
         public PluginResolver(INDMFPlatformProvider? platform = null, bool includeDisabled = false) : this(FindPluginTypes(), platform, includeDisabled)
         {
         }
 
         internal PluginResolver(IEnumerable<Type> plugins, INDMFPlatformProvider? platform, bool includeDisabled = false) 
-            : this(plugins.Select(InstantiatePlugin), platform, includeDisabled)
+            : this(plugins.Select(InstantiatePlugin).Where(p => p != null)!, platform, includeDisabled)
```

Comment:
When explicitly passing `Type`, it should be an error for `InstantiatePlugin` to return null.